kernel/src/hil/radio.rs

```diff
@@ -102,12 +104,81 @@ pub trait RadioConfig {
 
 pub trait RadioData {
     fn set_transmit_client(&self, client: &'static TxClient);
-    fn set_receive_client(&self, client: &'static RxClient, receive_buffer: &'static mut [u8]);
+    fn set_receive_client(&self, client: &'static RxClient, buffer: &'static mut [u8]);
     fn set_receive_buffer(&self, receive_buffer: &'static mut [u8]);
 
     fn transmit(
         &self,
-        spi_buf: &'static mut [u8],
-        frame_len: usize,
+        buf: &'static mut [u8],
+        len: usize,
     ) -> (ReturnCode, Option<&'static mut [u8]>);
 }
+
+#[derive(PartialEq, Debug, Copy, Clone)]
+pub enum RadioChannel {
```

Comment:
The enum is good, but I don't think these values should be in the HIL: it looks to be an NRF specific thing? Or are these frequencies (2.4xx MHz)? What do those values mean?

Reply:
Those are introduced first as `BLE radio channels` IIRC, i.e, 2400MHz-2480MHz

Reply:
I was going off the ble_advertising.rs file in the kernel/src directory. It also defines a `RadioChannel` enum that corresponds to the value being put in the FREQUENCY register.